### ISSUE TITLE
tRPC failed

### ISSUE THREAD

**Issue by janareddyc7** [quenti-io/quenti#266]
tRPC failed on recent.get: UNAUTHORIZED@quenti/next:dev: ❌ tse the ref parameter?
RPC failed on recent.get: UNAUTHORIZED

do you know how to fix it

**Comment by janareddyc7** [comment]
when i try to go to dashboard it says tRPC failed and routes me to login page

**Comment by miapolis** [comment]
I think I know what your issue might be. If you set up your oauth credentials properly, make sure you have a mysql database running and run `yarn prisma db:push` from the root directory. If NextAuth can't save your credentials to db it'll always throw unauthorized.

**Comment by janareddyc7** [comment]
i use planetscale i already did that let me try again

**Comment by miapolis** [comment]
> Also why the env folder in packages contains a lot optional things

There's a lot of other services Quenti relies on that _technically_ won't prevent the app from starting. Cohere, Huggingface, R2, Resend and others are all optional but you'll have trouble using parts of the app that rely on them.
https://github.com/quenti-io/quenti/issues/222#issuecomment-1793790236

**Comment by janareddyc7** [comment]
i think hugginface is paid

**Comment by miapolis** [comment]
> i think hugginface is paid

Right, which is why Quenti isn't all that useful in a local environment unless you're willing to pay all of the infra costs

**Comment by janareddyc7** [comment]
thanks

**Comment by janareddyc7** [comment]
what should the env values be for these
SERVER_NAME
 USERS_BUCKET_NAME: 
    USERS_BUCKET_URL: 
    ASSETS_BUCKET_NAME: 
    ASSETS_BUCKET_URL:
ENABLE_EMAIL_WHITELIST:

**Comment by janareddyc7** [comment]
GRAFANA_DASHBOARD_URL:
    STRIPE_ORG_MONTHLY_PRICE_ID:
CLICKHOUSE_HOST:
    CLICKHOUSE_USER:
    CLICKHOUSE_PASSWORD

**Comment by janareddyc7** [comment]
where do you get thesse values

**Comment by miapolis** [comment]
Currently undocumented, I'll address these briefly and close the issue since this is out of scope. You can create another issue if you'd like to request documentation, or open a discussion.

`SERVER_NAME` - used to distinguish between production and staging envs on vercel 
`USERS_BUCKET_NAME` - the name of the Cloudflare R2 bucket used for storing user assets (profile pictures, etc.)
`USERS_BUCKET_URL` - url of the users bucket
`ASSETS_BUCKET_NAME` - name of the r2 bucket for assets (org images, classes, term assets)
`ASSETS_BUCKET_URL` - url of assets bucket
`ENABLE_EMAIL_WHITELIST` - optionally enable an email whitelist to prevent external signups
`GRAFANA_DASHBOARD_URL` - grafana dashboard url
`STRIPE_ORG_MONTHLY_PRICE_ID` - stripe price id for a monthly subscription
all of the clickhouse env vars should be self explanatory
